src/JsonService.ts

```diff
@@ -27,6 +27,30 @@ export class JsonService {
         }
     }
 
+    protected async fetchWithTimeout(input: RequestInfo, init: RequestInit & { timeoutInSeconds?: number } = {}) {
+        const { timeoutInSeconds, ...initFetch } = init;
+        if (!timeoutInSeconds) {
+            return await fetch(input, initFetch);
+        }
+
+        try {
+            const controller = new AbortController();
+            const timeoutId = setTimeout(() => controller.abort(), timeoutInSeconds * 1000);
+            const response = await fetch(input, {
+                ...init,
+                signal: controller.signal,
+            });
+            clearTimeout(timeoutId);
+            return response;
+        }
+        catch (err) {
+            if ((err as Error).name === "AbortError") {
```

Comment:
I think this is fine, though you can make this a tad stricter with `if (err instanceof DOMException && err.name === 'AbortError')`.